velox/core/PlanNode.h

```diff
@@ -1751,6 +1753,14 @@ class TopNNode : public PlanNode {
         "Number of sorting keys and sorting orders in TopN must be the same");
     VELOX_USER_CHECK_GT(
         count, 0, "TopN must specify greater than zero number of rows to keep");
+    folly::F14FastSet<std::string> sortingKeyNameSet;
+    for (const auto& sortingKey : sortingKeys_) {
+      const auto& keyName = sortingKey->name();
+      if (sortingKeyNameSet.contains(keyName)) {
```

Comment:
- Let' move this implementation to .cpp file.
- Consider checking the results of 'insert' instead of an extra lookup and clarify that duplicate sorting key was found in TopN node:

```
bool ok = sortingKeyNameSet.insert(keyName).second;
VELOX_USER_CHECK(ok, "TopN node doesn't allow duplicate sorting keys: {}")
```
- Finally, add a test. See https://github.com/facebookincubator/velox/pull/7152 and https://github.com/facebookincubator/velox/pull/7210 for examples.